lib/stitch_fix/log_weasel/middleware.rb

```diff
@@ -14,16 +15,23 @@ def initialize(app, options = {})
     # Future: Maybe add X-Amzn-Trace-Id request header for tracing through load balancer
     # (http://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-request-tracing.html)
     def call(env)
+      req = Rack::Request.new(env)
       x_correlation_id = env.fetch(REQUEST_ID_KEY, nil) || env.fetch(CORRELATION_ID_KEY, nil)
 
+      if ENV.fetch('LOGWEASEL_FROM_PARAMS', nil) && req.params[PARAMS_KEY]
+        x_correlation_id = req.params[PARAMS_KEY]
+      end
+
       if x_correlation_id
         LogWeasel::Transaction.id = x_correlation_id
       else
         log_weasel_key = env.fetch("HTTP_#{KEY_HEADER}", @key)
         LogWeasel::Transaction.create(log_weasel_key)
-        env[CORRELATION_ID_KEY] = LogWeasel::Transaction.id
-        env[REQUEST_ID_KEY] = LogWeasel::Transaction.id
       end
+
+      env[CORRELATION_ID_KEY] = LogWeasel::Transaction.id
+      env[REQUEST_ID_KEY] = LogWeasel::Transaction.id
```

Comment:
I cannot think of a reason we would not want to set these each time.  The way `logger` currently works is it pulls the `uuid`/`request_id` from the controller request object - which if `HTTP_X_REQUEST_ID` is nil it assigns a new UUID.  So if we want to be able to correlate logs when `HTTP_X_REQUEST_ID` isn't set (i.e. when pulling this from params) we could potentially just _always_ set this.

Reply:
Yeah, with the introduction of support for passing the log weasel ID as a query param, I think you're right. Prior, these essentially would always be set anyway - either they'd come in on the request, or we'd create a new LWID and set them on the request. It feels a tiny bit icky to overwrite those headers with the same value in the former case, but I can live with it. 

I love this though ... thanks for taking the time to put this together! One nit: would you be open to creating a separate method for the params parsing bit? I'm thinking something like,
```
log_weasel_id_from_params(req)
```
and tacking an `||` onto the end of line 19?

Reply:
I believe I captured the spirit of this in https://github.com/stitchfix/log_weasel/pull/25/commits/c812d840b42125528452b3ecda6b4c5bb8389b58

Also, added a spec to check if the env var is on, but the request doesn't have the id in the params.